lib/rides/pages/search_deals_page.dart

```diff
@@ -0,0 +1,291 @@
+  final TextEditingController _destinationController = TextEditingController();
+
+  final _dateController = TextEditingController();
+  final _timeController = TextEditingController();
+  late final DateTime _firstDate;
+  late DateTime _selectedDate;
+  late int _dropdownValue;
+
+  final List<int> list = List.generate(10, (index) => index + 1);
+
+  List<Ride>? ridesuggestion;
+
+  @override
+  initState()  {
+    super.initState();
+    _firstDate = widget.date;
+    _selectedDate = widget.date;
+    _dateController.text = _formatDate(widget.date);
+    _timeController.text = _formatTime(widget.date);
+    _dropdownValue = widget.seats;
+    _startController.text = widget.start;
+    _destinationController.text = widget.end;
+    getRides();
+  }
+  @override
+  void dispose() {
+    _dateController.dispose();
+    _timeController.dispose();
+    _startController.dispose();
+    _destinationController.dispose();
+    super.dispose();
+  }
+
+  void _showDatePicker() {
+    showDatePicker(
+      context: context,
+      initialDate: _selectedDate,
+      firstDate: _firstDate,
+      lastDate: _firstDate.add(const Duration(days: 30)),
+    ).then((value) {
+      if (value != null) {
+        _selectedDate = DateTime(value.year, value.month, value.day,
+            _selectedDate.hour, _selectedDate.minute);
+        _dateController.text = _formatDate(_selectedDate);
+      }
+      getRides();
+    });
+  }
+
+  void _showTimePicker() {
+    showTimePicker(
+      context: context,
+      initialTime: TimeOfDay(hour: _selectedDate.hour, minute: _selectedDate.minute),
+      builder: (context, childWidget) {
+        return MediaQuery(data: MediaQuery.of(context).copyWith(alwaysUse24HourFormat: true), child: childWidget!);
+      },
+    ).then((value) {
+      setState(() {
+        if (value != null) {
+          _selectedDate =
+              DateTime(_selectedDate.year, _selectedDate.month, _selectedDate.day, value.hour, value.minute);
+          _timeController.text = _formatTime(_selectedDate);
+        }
+      });
+    });
+  }
+
+  String _formatDate(DateTime date) {
+    return '${date.day}.${date.month}.${date.year}';
+  }
+
+  String _formatTime(DateTime time) {
+    return DateFormat.Hm().format(time);
+  }
+
+  String? _timeValidator(String? value) {
+    if (value == null || value.isEmpty) {
+      return 'Please enter a time';
+    }
+    if (_selectedDate.isBefore(_firstDate)) {
+      return 'Please enter a valid time';
+    }
+    return null;
+  }
+
+  //todo: get possible Rides from Algorithm
+  void getRides() async {
+    List<Ride> initializes = await supabaseClient
+        .from('drives').select()
+        .eq('start', _startController.text)
+        .order('start_time', ascending: true)
+        .then((drive)
+    => Drive.fromJsonList(drive).map((e)
+    => e.toRide(_startController.text,
+      _destinationController.text,
+      e.startTime,
+      e.endTime,
+      _dropdownValue,
+      -1,
```

Comment:
I think it would be better if this was the userId if the user is logged in and only -1 when no user is logged in.

Reply:
Mh ja verstehe. Fand halt das man so alle Search Angebote einfach identifizieren kann. Da nur Angebote -1 haben können und alle korrekten Rides haben dann ein validen wert. Aber kann man naklar auch ändern.

Reply:
Ja das stimmt zwar aber die braucht man ja eh nur auf der einen Seite und auf dem Detailview denke nicht dass das ein Problem ist.